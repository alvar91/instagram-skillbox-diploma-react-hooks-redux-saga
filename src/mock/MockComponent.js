import React from "react";

function MockComponent() {
  return <h2>Mock Component</h2>;
}

export default MockComponent;
