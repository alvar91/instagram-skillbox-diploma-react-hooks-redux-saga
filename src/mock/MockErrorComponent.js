import React from "react";

function MockErrorComponent() {
  React.useEffect(() => {
    throw new Error("Mock error");
  }, []);

  return <h2>Mock error</h2>;
}

export default MockErrorComponent;
