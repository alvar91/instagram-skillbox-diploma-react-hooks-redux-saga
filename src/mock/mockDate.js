export const mockDate = "2021-02-11T17:25:19-05:00";
export const correctMockDate = "Fri, Feb 12, 2021";
