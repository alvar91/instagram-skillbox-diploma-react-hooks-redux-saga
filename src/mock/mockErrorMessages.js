export const mockAuthErrorMessage =
  "Failed to get access token. Please try to log in again";
export const mockErrorMeta = "meta error";
