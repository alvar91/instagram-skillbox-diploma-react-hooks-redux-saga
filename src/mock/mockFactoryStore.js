/**
 * Function to create a mock store with input options
 * @param {object} storeOptionsObj contains options to create the mock store
 * @returns {object} mock store
 */
export const mockFactoryStore = ({postsObj = {}, usersObj = {}, authObj = {}}) => {
  const {
    posts = [],
    currentPage = 1,
    isFetchingPost = false,
    hasErrorPosts = false,
    errorMessagePosts = "",
  } = postsObj;

  const { users = [] } = usersObj;

  const {
    isFetchingAuth = false,
    isLoggedIn = false,
    profile = [],
    hasErrorAuth = false,
    errorMessageAuth = "",
  } = authObj;

  return {
    /* posts state */
    posts: {
      isFetchingPost: isFetchingPost,
      currentPage: currentPage,
      posts: posts,
      users: users,
      hasError: hasErrorPosts,
      errorMessage: errorMessagePosts,
    },
    /* auth state */
    auth: {
      isFetchingAuth: isFetchingAuth,
      isLoggedIn: isLoggedIn,
      profile: profile,
      hasError: hasErrorAuth,
      errorMessage: errorMessageAuth,
    },
  };
};
