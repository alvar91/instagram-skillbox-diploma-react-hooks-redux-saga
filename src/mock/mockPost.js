export const mockPost = {
  id: "1gLdTsX3_70",
  created_at: "2020-07-01T18:30:14-04:00",
  updated_at: "2021-02-11T23:12:42-05:00",
  promoted_at: null,
  width: 6016,
  height: 4016,
  color: "#260c0c",
  blur_hash: "LHEf7wbbM{w[_NM{-:xZRiVs%gNG",
  description: null,
  alt_description: "laptop on brown sofa",
  urls: {
    raw:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1",
    full:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
    regular:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
    small:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
    thumb:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
  },
  links: {
    self: "https://api.unsplash.com/photos/1gLdTsX3_70",
    html: "https://unsplash.com/photos/1gLdTsX3_70",
    download: "https://unsplash.com/photos/1gLdTsX3_70/download",
    download_location: "https://api.unsplash.com/photos/1gLdTsX3_70/download",
  },
  categories: [],
  likes: 738,
  liked_by_user: false,
  current_user_collections: [],
  sponsorship: {
    impression_urls: [
      "https://secure.insightexpressai.com/adServer/adServerESI.aspx?script=false&bannerID=8281535&rnd=[timestamp]&gdpr=&gdpr_consent=&redir=https://secure.insightexpressai.com/adserver/1pixel.gif",
    ],
    tagline: "Designed to be the Best",
    tagline_url: "http://www.dell.com/xps",
    sponsor: {
      id: "2DC3GyeqWjI",
      updated_at: "2021-02-12T14:33:51-05:00",
      username: "xps",
      name: "XPS",
      first_name: "XPS",
      last_name: null,
      twitter_username: "Dell",
      portfolio_url: "http://www.dell.com/xps",
      bio:
        "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
      location: null,
      links: {
        self: "https://api.unsplash.com/users/xps",
        html: "https://unsplash.com/@xps",
        photos: "https://api.unsplash.com/users/xps/photos",
        likes: "https://api.unsplash.com/users/xps/likes",
        portfolio: "https://api.unsplash.com/users/xps/portfolio",
        following: "https://api.unsplash.com/users/xps/following",
        followers: "https://api.unsplash.com/users/xps/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "dell",
      total_collections: 0,
      total_likes: 0,
      total_photos: 22,
      accepted_tos: true,
    },
  },
  user: {
    id: "2DC3GyeqWjI",
    updated_at: "2021-02-12T14:33:51-05:00",
    username: "xps",
    name: "XPS",
    first_name: "XPS",
    last_name: null,
    twitter_username: "Dell",
    portfolio_url: "http://www.dell.com/xps",
    bio:
      "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
    location: null,
    links: {
      self: "https://api.unsplash.com/users/xps",
      html: "https://unsplash.com/@xps",
      photos: "https://api.unsplash.com/users/xps/photos",
      likes: "https://api.unsplash.com/users/xps/likes",
      portfolio: "https://api.unsplash.com/users/xps/portfolio",
      following: "https://api.unsplash.com/users/xps/following",
      followers: "https://api.unsplash.com/users/xps/followers",
    },
    profile_image: {
      small:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
      medium:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
      large:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
    },
    instagram_username: "dell",
    total_collections: 0,
    total_likes: 0,
    total_photos: 22,
    accepted_tos: true,
  },
};
