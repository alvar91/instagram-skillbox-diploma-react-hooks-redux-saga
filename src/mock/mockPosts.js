export const mockPosts = [
  {
    id: "1gLdTsX3_70",
    created_at: "2020-07-01T18:30:14-04:00",
    updated_at: "2021-02-11T23:12:42-05:00",
    promoted_at: null,
    width: 6016,
    height: 4016,
    color: "#260c0c",
    blur_hash: "LHEf7wbbM{w[_NM{-:xZRiVs%gNG",
    description: null,
    alt_description: "laptop on brown sofa",
    urls: {
      raw:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1",
      full:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
    },
    links: {
      self: "https://api.unsplash.com/photos/1gLdTsX3_70",
      html: "https://unsplash.com/photos/1gLdTsX3_70",
      download: "https://unsplash.com/photos/1gLdTsX3_70/download",
      download_location: "https://api.unsplash.com/photos/1gLdTsX3_70/download",
    },
    categories: [],
    likes: 738,
    liked_by_user: false,
    current_user_collections: [],
    sponsorship: {
      impression_urls: [
        "https://secure.insightexpressai.com/adServer/adServerESI.aspx?script=false&bannerID=8281535&rnd=[timestamp]&gdpr=&gdpr_consent=&redir=https://secure.insightexpressai.com/adserver/1pixel.gif",
      ],
      tagline: "Designed to be the Best",
      tagline_url: "http://www.dell.com/xps",
      sponsor: {
        id: "2DC3GyeqWjI",
        updated_at: "2021-02-12T14:33:51-05:00",
        username: "xps",
        name: "XPS",
        first_name: "XPS",
        last_name: null,
        twitter_username: "Dell",
        portfolio_url: "http://www.dell.com/xps",
        bio:
          "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
        location: null,
        links: {
          self: "https://api.unsplash.com/users/xps",
          html: "https://unsplash.com/@xps",
          photos: "https://api.unsplash.com/users/xps/photos",
          likes: "https://api.unsplash.com/users/xps/likes",
          portfolio: "https://api.unsplash.com/users/xps/portfolio",
          following: "https://api.unsplash.com/users/xps/following",
          followers: "https://api.unsplash.com/users/xps/followers",
        },
        profile_image: {
          small:
            "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
          medium:
            "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
          large:
            "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
        },
        instagram_username: "dell",
        total_collections: 0,
        total_likes: 0,
        total_photos: 22,
        accepted_tos: true,
      },
    },
    user: {
      id: "2DC3GyeqWjI",
      updated_at: "2021-02-12T14:33:51-05:00",
      username: "xps",
      name: "XPS",
      first_name: "XPS",
      last_name: null,
      twitter_username: "Dell",
      portfolio_url: "http://www.dell.com/xps",
      bio:
        "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
      location: null,
      links: {
        self: "https://api.unsplash.com/users/xps",
        html: "https://unsplash.com/@xps",
        photos: "https://api.unsplash.com/users/xps/photos",
        likes: "https://api.unsplash.com/users/xps/likes",
        portfolio: "https://api.unsplash.com/users/xps/portfolio",
        following: "https://api.unsplash.com/users/xps/following",
        followers: "https://api.unsplash.com/users/xps/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "dell",
      total_collections: 0,
      total_likes: 0,
      total_photos: 22,
      accepted_tos: true,
    },
  },
  {
    id: "lqFM73nWnf8",
    created_at: "2021-02-12T05:41:16-05:00",
    updated_at: "2021-02-12T06:17:12-05:00",
    promoted_at: "2021-02-12T06:04:16-05:00",
    width: 5760,
    height: 6121,
    color: "#595959",
    blur_hash: "LD7d%rt7ofof00Rjayay-;j[j[ay",
    description: "Architecture abstract",
    alt_description: "gray scale photo of palm tree",
    urls: {
      raw:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1",
      full:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
    },
    links: {
      self: "https://api.unsplash.com/photos/lqFM73nWnf8",
      html: "https://unsplash.com/photos/lqFM73nWnf8",
      download: "https://unsplash.com/photos/lqFM73nWnf8/download",
      download_location: "https://api.unsplash.com/photos/lqFM73nWnf8/download",
    },
    categories: [],
    likes: 31,
    liked_by_user: false,
    current_user_collections: [],
    sponsorship: null,
    user: {
      id: "r01XoGoEPT8",
      updated_at: "2021-02-12T14:13:55-05:00",
      username: "ripato",
      name: "Ricardo Gomez Angel",
      first_name: "Ricardo",
      last_name: "Gomez Angel",
      twitter_username: null,
      portfolio_url: "https://www.flickr.com/photos/rigoan/",
      bio:
        "I'm architect, marketing specialist and photographer. I`m working since 2 years as freelancer photographer for different Swiss companies",
      location: "Switzerland",
      links: {
        self: "https://api.unsplash.com/users/ripato",
        html: "https://unsplash.com/@ripato",
        photos: "https://api.unsplash.com/users/ripato/photos",
        likes: "https://api.unsplash.com/users/ripato/likes",
        portfolio: "https://api.unsplash.com/users/ripato/portfolio",
        following: "https://api.unsplash.com/users/ripato/following",
        followers: "https://api.unsplash.com/users/ripato/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1492187901140-48ac7eff801c?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1492187901140-48ac7eff801c?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1492187901140-48ac7eff801c?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "ricardogo03",
      total_collections: 0,
      total_likes: 14717,
      total_photos: 1159,
      accepted_tos: true,
    },
  },
  {
    id: "W3tpb2wGRfo",
    created_at: "2021-02-11T10:13:29-05:00",
    updated_at: "2021-02-12T06:17:12-05:00",
    promoted_at: "2021-02-12T05:12:04-05:00",
    width: 2277,
    height: 2664,
    color: "#73a6a6",
    blur_hash: "LJI}|o0fCm^+BWozE1WVK6xa-VNG",
    description: null,
    alt_description: "white concrete building under blue sky during daytime",
    urls: {
      raw:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1",
      full:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=80&w=200",
    },
    links: {
      self: "https://api.unsplash.com/photos/W3tpb2wGRfo",
      html: "https://unsplash.com/photos/W3tpb2wGRfo",
      download: "https://unsplash.com/photos/W3tpb2wGRfo/download",
      download_location: "https://api.unsplash.com/photos/W3tpb2wGRfo/download",
    },
    categories: [],
    likes: 52,
    liked_by_user: false,
    current_user_collections: [],
    sponsorship: null,
    user: {
      id: "E37oh-Tr6bo",
      updated_at: "2021-02-12T14:13:54-05:00",
      username: "bennmcguinness",
      name: "Benn McGuinness",
      first_name: "Benn",
      last_name: "McGuinness",
      twitter_username: null,
      portfolio_url: null,
      bio: null,
      location: "London ",
      links: {
        self: "https://api.unsplash.com/users/bennmcguinness",
        html: "https://unsplash.com/@bennmcguinness",
        photos: "https://api.unsplash.com/users/bennmcguinness/photos",
        likes: "https://api.unsplash.com/users/bennmcguinness/likes",
        portfolio: "https://api.unsplash.com/users/bennmcguinness/portfolio",
        following: "https://api.unsplash.com/users/bennmcguinness/following",
        followers: "https://api.unsplash.com/users/bennmcguinness/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1611603138308-604e48efa757image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1611603138308-604e48efa757image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1611603138308-604e48efa757image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "benn.mcguinness",
      total_collections: 1,
      total_likes: 1583,
      total_photos: 448,
      accepted_tos: true,
    },
  },
];
