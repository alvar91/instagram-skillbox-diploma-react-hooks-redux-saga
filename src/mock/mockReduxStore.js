// Core
import { createStore, applyMiddleware } from "redux";

// Instruments
import { rootReducer } from "../redux/rootReducer";
import {
  composeEnhancers,
  middleware,
  sagaMiddleware,
} from "../redux/middleware";
import { rootSaga } from "../redux/rootSaga";

// Mock state
import { mockFactoryStore } from "./mockFactoryStore";

export const mockReduxStore = (storeOptionsObj) => {
  const initialState = mockFactoryStore(storeOptionsObj);
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(...middleware))
  );

  sagaMiddleware.run(rootSaga);
  return store;
};
