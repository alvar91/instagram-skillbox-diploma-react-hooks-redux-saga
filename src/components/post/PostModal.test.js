// Core
import React from "react";
import { configure, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import PostModal from "./PostModal";
import { mockPost } from "../../mock/mockPost";
import { mockPosts } from "../../mock/mockPosts";
import { MemoryRouter, Route } from "react-router-dom";

configure({ adapter: new Adapter() });

// Instruments
import { Provider } from "react-redux";
import { mockReduxStore } from "../../mock/mockReduxStore";

/**
 * Setup function for FeedPost component.
 * @param {object} props
 * @returns {ReactRenderer}
 */
const setupTestRenderer = (storeOptionsObj) => {
    const mockStore = mockReduxStore(storeOptionsObj);
  return mount(
    <Provider store={mockStore}>
      <MemoryRouter
        initialEntries={[ { pathname: `/${mockPost.id}`, key: 'testKey' } ]}
      >
        <Route path="/:postId">
          <PostModal />
        </Route>
      </MemoryRouter>
    </Provider>
  );
};

describe("FeedPost", () => {
  let testRenderer;
  let component;

  beforeAll(() => {
    const storeOptionsObj = {
      postsObj: {
        posts: mockPosts,
        currentPage: 1,
        isFetchingPost: false,
        hasErrorPosts: false,
        errorMessagePosts: "",
      },
      usersObj: {
        users: [],
      },
      authObj: {
        isFetchingAuth: false,
        isLoggedIn: false,
        profile: [],
        hasErrorAuth: false,
        errorMessageAuth: "",
      },
    };
    testRenderer = setupTestRenderer(storeOptionsObj);
    component = testRenderer.find(PostModal);
  });

  it("FeedPost should correctly renders markup", () => {
    expect(shallowToJson(testRenderer)).toMatchSnapshot();
  });
});
