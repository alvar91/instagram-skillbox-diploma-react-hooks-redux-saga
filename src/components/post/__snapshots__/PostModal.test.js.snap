// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedPost FeedPost should correctly renders markup 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/1gLdTsX3_70",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/1gLdTsX3_70",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/1gLdTsX3_70",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        path="/:postId"
      >
        <PostModal>
          <Modal
            ariaHideApp={false}
            bodyOpenClassName="ReactModal__Body--open"
            closeTimeoutMS={0}
            contentElement={[Function]}
            isOpen={true}
            onRequestClose={[Function]}
            overlayClassName="makeStyles-overlay-1"
            overlayElement={[Function]}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            preventScroll={false}
            role="dialog"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={true}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
            style={
              Object {
                "content": Object {
                  "WebkitOverflowScrolling": "touch",
                  "alignItems": "center",
                  "bottom": "auto",
                  "display": "flex",
                  "left": "50%",
                  "margin": 0,
                  "maxWidth": 935,
                  "overflow": "none",
                  "padding": 0,
                  "right": "auto",
                  "top": "50%",
                  "transform": "translate(-50%, -50%)",
                  "width": "100%",
                },
              }
            }
          >
            <Portal
              containerInfo={
                <div
                  class="ReactModalPortal"
                >
                  <div
                    class="ReactModal__Overlay ReactModal__Overlay--after-open makeStyles-overlay-1"
                  >
                    <div
                      aria-modal="true"
                      class="ReactModal__Content ReactModal__Content--after-open"
                      role="dialog"
                      style="position: absolute; top: 50%; left: 50%; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: none; border-radius: 4px; outline: none; padding: 0px; display: flex; align-items: center; max-width: 935px; width: 100%; transform: translate(-50%, -50%); margin: 0px;"
                      tabindex="-1"
                    >
                      <div
                        class="makeStyles-container-29"
                      >
                        <div
                          class="makeStyles-mediaSkeleton-35"
                        />
                        <div>
                          <div
                            class="makeStyles-headerSkeleton-30"
                          >
                            <div
                              class="makeStyles-avatarSkeleton-31"
                            />
                            <div
                              class="makeStyles-headerTextSkeleton-32"
                            >
                              <div
                                class="makeStyles-primaryTextSkeleton-33"
                              />
                              <div
                                class="makeStyles-secondaryTextSkeleton-34"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              }
            >
              <ModalPortal
                ariaHideApp={false}
                bodyOpenClassName="ReactModal__Body--open"
                closeTimeoutMS={0}
                contentElement={[Function]}
                defaultStyles={
                  Object {
                    "content": Object {
                      "WebkitOverflowScrolling": "touch",
                      "background": "#fff",
                      "border": "1px solid #ccc",
                      "borderRadius": "4px",
                      "bottom": "40px",
                      "left": "40px",
                      "outline": "none",
                      "overflow": "auto",
                      "padding": "20px",
                      "position": "absolute",
                      "right": "40px",
                      "top": "40px",
                    },
                    "overlay": Object {
                      "backgroundColor": "rgba(255, 255, 255, 0.75)",
                      "bottom": 0,
                      "left": 0,
                      "position": "fixed",
                      "right": 0,
                      "top": 0,
                    },
                  }
                }
                isOpen={true}
                onRequestClose={[Function]}
                overlayClassName="makeStyles-overlay-1"
                overlayElement={[Function]}
                parentSelector={[Function]}
                portalClassName="ReactModalPortal"
                preventScroll={false}
                role="dialog"
                shouldCloseOnEsc={true}
                shouldCloseOnOverlayClick={true}
                shouldFocusAfterRender={true}
                shouldReturnFocusAfterClose={true}
                style={
                  Object {
                    "content": Object {
                      "WebkitOverflowScrolling": "touch",
                      "alignItems": "center",
                      "bottom": "auto",
                      "display": "flex",
                      "left": "50%",
                      "margin": 0,
                      "maxWidth": 935,
                      "overflow": "none",
                      "padding": 0,
                      "right": "auto",
                      "top": "50%",
                      "transform": "translate(-50%, -50%)",
                      "width": "100%",
                    },
                  }
                }
              >
                <div
                  className="ReactModal__Overlay ReactModal__Overlay--after-open makeStyles-overlay-1"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  style={Object {}}
                >
                  <div
                    aria-modal={true}
                    className="ReactModal__Content ReactModal__Content--after-open"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    role="dialog"
                    style={
                      Object {
                        "WebkitOverflowScrolling": "touch",
                        "alignItems": "center",
                        "background": "#fff",
                        "border": "1px solid #ccc",
                        "borderRadius": "4px",
                        "bottom": "auto",
                        "display": "flex",
                        "left": "50%",
                        "margin": 0,
                        "maxWidth": 935,
                        "outline": "none",
                        "overflow": "none",
                        "padding": 0,
                        "position": "absolute",
                        "right": "auto",
                        "top": "50%",
                        "transform": "translate(-50%, -50%)",
                        "width": "100%",
                      }
                    }
                    tabIndex="-1"
                  >
                    <Memo(Post)
                      id="1gLdTsX3_70"
                    >
                      <PostSkeleton>
                        <div
                          className="makeStyles-container-29"
                          style={
                            Object {
                              "gridTemplateColumns": false,
                            }
                          }
                        >
                          <div
                            className="makeStyles-mediaSkeleton-35"
                          />
                          <div>
                            <div
                              className="makeStyles-headerSkeleton-30"
                            >
                              <div
                                className="makeStyles-avatarSkeleton-31"
                              />
                              <div
                                className="makeStyles-headerTextSkeleton-32"
                              >
                                <div
                                  className="makeStyles-primaryTextSkeleton-33"
                                />
                                <div
                                  className="makeStyles-secondaryTextSkeleton-34"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </PostSkeleton>
                    </Memo(Post)>
                  </div>
                </div>
              </ModalPortal>
            </Portal>
          </Modal>
          <div
            className="makeStyles-close-2"
            onClick={[Function]}
          >
            <CloseIcon>
              <svg
                aria-label="Close"
                className="_8-yf5"
                fill="#fff"
                height={24}
                viewBox="0 0 48 48"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M41.8 9.8L27.5 24l14.2 14.2c.6.6.6 1.5 0 2.1l-1.4 1.4c-.6.6-1.5.6-2.1 0L24 27.5 9.8 41.8c-.6.6-1.5.6-2.1 0l-1.4-1.4c-.6-.6-.6-1.5 0-2.1L20.5 24 6.2 9.8c-.6-.6-.6-1.5 0-2.1l1.4-1.4c.6-.6 1.5-.6 2.1 0L24 20.5 38.3 6.2c.6-.6 1.5-.6 2.1 0l1.4 1.4c.6.6.6 1.6 0 2.2z"
                  fillRule="evenodd"
                />
              </svg>
            </CloseIcon>
          </div>
        </PostModal>
      </Route>
    </Router>
  </MemoryRouter>
</Provider>
`;
